src/pcre2_jit_compile.c

```diff
@@ -2460,7 +2461,7 @@ status->compiler = common->compiler;
 }
 
 static void delayed_mem_copy_move(delayed_mem_copy_status *status, int load_base, sljit_sw load_offset,
-  int store_base, sljit_sw store_offset)
+  int store_base, int store_offset)
```

Comment:
The `store_offset` arg is placed in the `status->store_offsets` array, which is already an `int` in the current code. So I think it's safe to make the argument an int (and hence avoid truncation when assigning to `status->store_offsets`).

That just moves the problem elsewhere though: now we need to worry about truncating to a 32-bit int in all the code which _calls_ `delayed_mem_copy_move`.